weather_app_functions.R

```diff
@@ -0,0 +1,234 @@
+  
+  my_content_from_json <- jsonlite::fromJSON(my_content)
+  
+  return(my_content_from_json)
+  
+}
+
+## Example for get_weather 
+get_weather("Amsterdam, Niederlande", Sys.getenv("MY_API"))
+
+
+# This function gets as input whether the user wants the current weather forecast, the hourly forecast, 
+# or the daily forecast, the location of interest and the apiKey. It returns a list with the desired wheather information.
+
+get_your_forecast <- function(cur_hour_day = current, location, apiKey) {
+  
+  full_weather <- get_weather(location, apiKey)
+  
+  if (cur_hour_day == "current") {
+    
+    weather <- full_weather$current 
+    
+  } else if (cur_hour_day == "hourly") {
+    
+    weather <- full_weather$hourly
+    
+  } else if (cur_hour_day == "daily") {
+    
+    weather <- full_weather$daily
+    
+  }
+  
+  return(weather)
+  
+}
+
+## Example for yourForecast()
+get_your_forecast("hourly", "Amsterdam, Niederlande", sys.getenv(MY_API))
+  
+
+# This function get a location (as a string) as input and returns and saves a map of this location using leaflet::leaflet().  
+
+get_map <- function(location) {
+  
+  latitude <- tidygeocoder::geo_osm(location)$lat[1]
+  longitude <- tidygeocoder::geo_osm(location)$long[1]
+  
+  map <- leaflet::leaflet() %>% 
+    leaflet::setView(lng = longitude, lat = latitude, zoom = 11) %>%
+    leaflet::addProviderTiles(providers$Stamen.TonerLite, options = providerTileOptions(noWrap = TRUE))
+  
+  mapview::mapshot(map, file = "App-1/Rplot.png")
+  map <- image_read("App-1/Rplot.png")
+  
+  return(map)
+  
+}
+
+## Example for get_map()
+get_map("Amsterdam, Niederlande")
+
+
+# This function gets a location and an apiKey as input. It looks for the current weather for the specified location and 
+# saves and returns a map of the location.
+# The map includes icons that correspond to the current weather forecast for this location. 
+
+get_icon_map <- function(location, apiKey) {
+  
+  latitude <- tidygeocoder::geo_osm(location)$lat[1]
+  longitude <- tidygeocoder::geo_osm(location)$long[1]
+  
+  map <- leaflet() %>% 
+    setView(lng = longitude, lat = latitude, zoom = 12) %>%
+    addProviderTiles(providers$Stamen.TonerLite, options = providerTileOptions(noWrap = TRUE))
+  mapshot(map, file = "App-1/Rplot.png")
+  
+  my_weather <- get_weather(location, apiKey)
+  
+  mymap <- image_scale(image_read(path = "App-1/Rplot.png"), "x400")
+  imageName <- paste("App-1/www/",my_weather$current$weather$icon,".png", sep = "")
+  icon <- image_scale(image_read("App-1/www/02d.png"), "x100")
+  
+  img <- c(mymap, icon)
+  myMapImage <- image_mosaic(img)
+  
+  image1 <- image_composite(mymap, icon, offset = "+20+20")
+  image2 <-image_composite(image1, icon, offset = "+280+180")
+  
+  image_write(myMapImage, path = "my_weather.png", format = "png")
+  image_write(image2, path = "my_weather2.png", format = "png")
+  
+  return(myMapImage)
+  
+}

```

Comment:
perhaps you could add some more comments in case the code is not self-commenting